src/nvim/drawline.c

```diff
@@ -165,12 +143,14 @@ void drawline_free_all_mem(void)
 /// Advance **color_cols
 ///
 /// @return  true when there are columns to draw.
-static bool advance_color_col(int vcol, int **color_cols)
+static void advance_color_col(winlinevars_T *wlv, int vcol)
 {
-  while (**color_cols >= 0 && vcol > **color_cols) {
-    (*color_cols)++;
+  if (wlv->draw_color_col) {
+    while (*wlv->color_cols >= 0 && vcol > *wlv->color_cols) {
```

Comment:
```suggestion
    while (*wlv->color_cols >= 0 && wlv->vcol > *wlv->color_cols) {
```

Reply:
I don't think this will be correct with the cases which used `wlv.vcol - wlv.vcol_off` (of course we want to try simplify that but that is for the next PR).